File: app/src/main/java/pl/revanmj/smspasswordnotifier/data/WhitelistRepository.java
package pl.revanmj.smspasswordnotifier.data;

import android.app.Application;
import android.arch.lifecycle.LiveData;
import android.os.AsyncTask;

import java.util.List;

public class WhitelistRepository {
    private WhitelistDao mWhitelistDao;
    private LiveData<List<WhitelistItem>> mWhitelist;

    WhitelistRepository(Application application) {
        WhitelistDb db = WhitelistDb.getDatabase(application);
        mWhitelistDao = db.getWhitelistDao();
        mWhitelist = mWhitelistDao.getAll();
    }

    LiveData<List<WhitelistItem>> getWhitelist() {
        return mWhitelist;
    }

    WhitelistItem getItemByName(String name) { return mWhitelistDao.getByName(name); }

    public void insert(WhitelistItem... items) {
        new insertAsyncTask(mWhitelistDao).execute(items);
    }

    public void delete(WhitelistItem... items) {
        new deleteAsyncTask(mWhitelistDao).execute(items);
    }

    private static class insertAsyncTask extends AsyncTask<WhitelistItem, Void, Void> {

        private WhitelistDao mAsyncTaskDao;

        insertAsyncTask(WhitelistDao dao) {
            mAsyncTaskDao = dao;
        }

        @Override
        protected Void doInBackground(final WhitelistItem... params) {
            mAsyncTaskDao.insertAll(params);
            return null;
        }
    }

    private static class deleteAsyncTask extends AsyncTask<WhitelistItem, Void, Void> {

        private WhitelistDao mAsyncTaskDao;

        deleteAsyncTask(WhitelistDao dao) {
            mAsyncTaskDao = dao;
        }

        @Override
        protected Void doInBackground(final WhitelistItem... params) {
            mAsyncTaskDao.deleteAll(params);
            return null;
        }
    }
}

File: app/src/main/java/pl/revanmj/smspasswordnotifier/data/WhitelistRepository.kt
package pl.revanmj.smspasswordnotifier.data

import android.app.Application

import androidx.lifecycle.LiveData

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class WhitelistRepository internal constructor(application: Application) {
    private val mWhitelistDao: WhitelistDao
    internal val whitelist: LiveData<List<WhitelistItem>>

    init {
        val db = WhitelistDb.getDatabase(application)
        mWhitelistDao = db.whitelistDao
        whitelist = mWhitelistDao.all
    }

    internal fun getItemByName(name: String): WhitelistItem {
        return mWhitelistDao.getByName(name)
    }

    fun insert(vararg items: WhitelistItem) {
        GlobalScope.launch(Dispatchers.IO) {
            mWhitelistDao.insertAll(*items)
        }
    }

    fun update(vararg items: WhitelistItem) {
        GlobalScope.launch(Dispatchers.IO) {
            mWhitelistDao.updateAll(*items)
        }
    }

    fun delete(vararg items: WhitelistItem) {
        GlobalScope.launch(Dispatchers.IO) {
            mWhitelistDao.deleteAll(*items)
        }
    }
}
